### PR TITLE
Implement workaround for Issue 8329

### PR DESCRIPTION
**Pull request by markandrus**
_Not ready to merge._

https://bugs.chromium.org/p/webrtc/issues/detail?id=8329

### CODE REVIEW

File: lib/util/sdp/index.js [hunk 1]
```diff
@@ -31,23 +41,28 @@ function createBLine(modifier, maxBitrate) {
...
+/**
+ * Create a Map from PTs to codec names for the given m= section.
+ * @param {string} mediaSection - The given m= section.
+ * @returns {Map<PT, Codec>} ptToCodecName
+ */
+function createPtToCodecName(mediaSection) {
+  return getPayloadTypesInMediaSection(mediaSection).reduce(function(ptToCodecName, pt) {
+    var rtpmapPattern = new RegExp('a=rtpmap:' + pt + ' ([^/]+)');
+    var matches = mediaSection.match(rtpmapPattern);
+    var codecName = matches ? matches[1].toLowerCase() : '';
```
Comment (manjeshbhargav) [inline lib/util/sdp/index.js]: @markandrus Why are we adding an empty string for the payload type whose codec name doesn't exist?

File: lib/util/sdp/issue8329.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+ * A Set with at least one element
+ * @typedef {Set} NonEmptySet
+ */
+
+/**
+ * Apply the workaround for Issue 8329 to an RTCSessionDescriptionInit.
+ * @param {RTCSessionDescriptionInit} description
+ * @returns {RTCSessionDescription} newDescription
+ */
+function workaround(description) {
```
Comment (manjeshbhargav) [inline lib/util/sdp/issue8329.js]: @markandrus Can we simplify this:
```javascript
function workaroud(description) {
  var descriptionInit = { type: description.type };
  if (description.type !== 'rollback') {
    descriptionInit.sdp = sdpWorkaround(description.sdp);
  }
return new RTCSessionDescription(descriptionInit);
}
```

File: lib/util/sdp/issue8329.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  var unassociatedRtxPts = Array.from(invalidRtxPts);
+
+  // NOTE(mroberts): We normalize to lowercase.
+  var knownCodecNames = ['h264', 'vp8', 'vp9'];
+  var unassociatedPts = knownCodecNames.reduce(function(unassociatedPts, codecName) {
+    var pts = codecNameToPts.get(codecName) || new Set();
+    return Array.from(pts).reduce(function(unassociatedPts, pt) {
+      return associatedPtToRtxPt.has(pt)
+        ? unassociatedPts
+        : unassociatedPts.add(pt);
```
Comment (manjeshbhargav) [inline lib/util/sdp/issue8329.js]: @markandrus Are you sure `Set#add()` returns the `Set`?

File: lib/util/sdp/index.js (hunk 1 above)
Comment (markandrus) [reply lib/util/sdp/index.js]: Only dynamic PTs are required to have "a=rtpmap" entries—there are some that do not require these attributes. I guess we could/should look up their names statically, though? We don't really need/use them, though. The Paper doc where I describe this mentions letting the `codecName` be the empty string, though.

File: lib/util/sdp/issue8329.js (hunk 2 above)
Comment (markandrus) [reply lib/util/sdp/issue8329.js]: Good idea 👍

File: lib/util/sdp/issue8329.js (hunk 3 above)
Comment (markandrus) [reply lib/util/sdp/issue8329.js]: Yes https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add

**Comment by manjeshbhargav** [comment]
@markandrus Couple of nitpicks, but looks good otherwise.

**Pull request merged**